### ISSUE TITLE
ENS resolver in Trezor Suite

### ISSUE THREAD

**Issue by Hannsek** [trezor/trezor-suite#8618]
When user type `something`**.eth** or **.box**, Suite will ask Blockbook for the address connected to the name and fill it inside the address field.

![Screenshot 2024-05-02 at 15 11 06](https://github.com/trezor/trezor-suite/assets/33235762/ecb66455-eced-44e3-85ee-38499b3bf39b)

Usage
1. Send form
2. Account name
3. Transaction history
4. Receive tab (? no confirmation with device)

**Comment by AdamSchinzel** [comment]
Good idea. I was looking and the current API of blockbook doesn't have that.

Ethereum name service (ENS) operates as a set of smart contracts with two principal components: registry and resolvers. It could be in blockbook but there are also pretty good libraries for that such as [wagmi](https://wagmi.sh/) (a collection of React Hooks) that has [useEnsResolver hook](https://wagmi.sh/react/hooks/useEnsResolver) that could be used for that.

Also in [ENS Front-End Design Guidelines](https://docs.ens.domains/dapp-developer-guide/front-end-design-guidelines) there is also another recommendation where to use ENS names besides inputs and that is when showing addresses in e. g. transaction history where we can show ENS names instead of almost full address. Yes we can keep both but ENS address can be shown as "primary" there.

Happy to be assigned on this.

**Comment by tomasklim** [comment]
We can also add it to transaction history as Blockbook returns those data.

It is in `account.misc` in Suite.
![Screenshot 2024-05-02 at 15 13 13](https://github.com/trezor/trezor-suite/assets/33235762/912578a0-1930-453f-998a-c92aa3686eb7)

It supports only ENS, not unstoppable domains, which we should also consider https://etherscan.io/name-lookup#

**Comment by tomasklim** [comment]
Waiting for https://github.com/trezor/blockbook/issues/1108

**Comment by 53gur0** [comment]
To propose a more decentralized approach to ENS resolution, I would refer to the original documentation: 

https://docs.ens.domains/web/subgraph#subgraph

**Comment by evgenysl** [comment]
@tomasklim 

https://github.com/trezor/blockbook/issues/1108 is completed
would be nice to have it in Suite now

**Comment by 53gur0** [comment]
QA done here: https://github.com/trezor/trezor-suite/pull/30402#issuecomment-5479740202
